processing/src/main/java/io/druid/query/extraction/CascadeExtractionFn.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.extraction;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Bytes;
+
+import java.util.Arrays;
+
+public class CascadeExtractionFn implements ExtractionFn
+{
+  private static final byte CACHE_TYPE_ID = 0x9;
+
+  private final ExtractionFn extractionFns[];
+  private ChainedExtractionFn chainedExtractionFn;
+
+  @JsonCreator
+  public CascadeExtractionFn(
+      @JsonProperty("extractionFns") ExtractionFn[] extractionFn
+  )
+  {
+    Preconditions.checkArgument(extractionFn != null, "extractionFns should not be null");
+    this.extractionFns = extractionFn;
+    ChainedExtractionFn root = null;
+    for (int idx = 0; idx < extractionFns.length; idx++) {
+      Preconditions.checkArgument(extractionFns[idx] != null, "empty function is not allowed");
+      root = new ChainedExtractionFn(extractionFns[idx], root);
+    }
+    this.chainedExtractionFn = root;  }
```

Comment:
this.chainedExtractionFn = (rool == null) ? DEFAULT_CHAINED_EXTRACTION_FN : root

also, minor formatting fix needed